src/components/menus/bus-menu.tsx

```diff
@@ -109,7 +113,7 @@ export const BusMenu: FunctionComponent<BusMenuProps> = ({
     const handleOpenDynamicSimulationEventDialog = useCallback(
         (equipmentId: string, equipmentType: string, dialogTitle: string) => {
             onClose();
-            onOpenDynamicSimulationEventDialog(equipmentId, equipmentType, dialogTitle);
+            onOpenDynamicSimulationEventDialog(equipmentId, equipmentType as EQUIPMENT_TYPES, dialogTitle);
```

Comment:
NOTE => do not cast, set type from parameter. 

you may need to change componant props too.
interface DynamicSimulationEventMenuItemProps<TequipementType extends string> {
    equipmentId: string;
    equipmentType: TequipementType;
    onOpenDynamicSimulationEventDialog: (equipmentId: string, equipmentType: TequipementType, dialogTitle: string) => void;
    disabled: boolean;
}